File: python/LeetCode2900.py
class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        result = []
        bit = -1
        for w, b in zip(words, groups):
            if b != bit:
                result.append(w)
                bit = b

        return result

File: cpp/LeetCode2900.cpp
class Solution {
public:
    vector<string> getLongestSubsequence(vector<string>& words, vector<int>& groups) {
        vector<string> result;
        int bit = -1;

        for (int i = 0; i < words.size(); i++){
            if (groups[i] != bit){
                result.push_back(words[i]);
                bit = groups[i];
            }
        }
        
        return result;
    }
};